java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -257,6 +257,9 @@ public static ProjectSource attachSource(String projectLabel, Type sourceType, S
                 // if a source has been DETACHED and we attach it again -> it gets back to original state (BUILT)
                 src.setState(BUILT);
             }
+            project.removeSource(src);
+            project.addSource(src, position);
```

Comment:
When `position` is empty, it'll "move" the source to the 1st position. Is that the intention here?

Reply:
I also think I'll need to adjust the javadoc - it says "when source has been already attached, this is a no-op".

Reply:
> When `position` is empty, it'll "move" the source to the 1st position. Is that the intention here?

Fixed in the last commit. It will check now if the position is present